src/esp/sim/Simulator.cpp

```diff
@@ -664,6 +666,21 @@ void Simulator::removeObject(const int objectID,
       // resourceManager_->removeResourceByName(trajVisAssetName);
     }
   }
+  // if (deleteObjectNode) {
+  //   // delete sensor at ObjectNode if there is one
+  //   std::map<std::string, esp::sensor::Sensor::ptr>::iterator sensorIterator
+  //   =
+  //       getSensorSuite().getSensors().find(std::to_string(objectID));
+  //   if (sensorIterator != getSensorSuite().getSensors().end()) {
+  //     //freeing memory at pointer to sensor
+  //     delete sensorIterator->second.get();
+  //     //erasing sensor from SensorSuite
+  //     getSensorSuite().getSensors().erase(sensorIterator);
```

Comment:
Probably don't want the delete in here.  By erasing sensorIterator from the map, you should be removing the last reference to the sensor's smart pointer, and it should clean up after itself.